lib/public/components/detector/PhysicalDetectorPickerModel.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { RemoteData } from '/js/src/index.js';
+import { detectorsProvider } from '../../services/detectors/detectorsProvider.js';
+import { PickerModel } from '../common/selection/picker/PickerModel.js';
+import { badge } from '../common/badge.js';
+
+/**
+ * Map detector to select options with the detector badge as label
+ *
+ * @param {Detector} detector detector
+ * @return {SelectOption} select option
+ */
+export const detectorToOption = ({ id, name }) => ({
```

Comment:
As this will be used for options of `h()` element, I think the `id` and `key` should be used as it is an iteration

Reply:
it's responsibility of picker , not picker model